### PR TITLE
[Firebase 10] Bump versions for 10.0.0

### PR DESCRIPTION
**Pull request by ncooke3**
### Context
- Bumped all Firebase products to 10.0.0
- Bumped dependencies on GTMSessionFetcher to min 2.1
- Bumped dependencies on GoogleUtilities to 7.8
- Bumped dependencies on GoogleDataTransport to 9.2

#### Follow-up tasks from #10233 
- Bumped tvOS min. support from 11.0 to 12.0
- Bumped AppCheck's min. support for iOS from 10.0 to 11.0

### CODE REVIEW

File: FirebaseAnalytics.podspec
Comment (ncooke3) [inline FirebaseAnalytics.podspec:35]:
```suggestion
    s.dependency 'nanopb', '>= 2.30908.0', '< 2.30910.0'
```

File: FirebaseCrashlytics.podspec
Comment (ncooke3) [inline FirebaseCrashlytics.podspec:61]:
```suggestion
  s.dependency 'nanopb', '>= 2.30908.0', '< 2.30910.0'
```

File: FirebaseInAppMessaging.podspec
Comment (ncooke3) [inline FirebaseInAppMessaging.podspec:85]:
```suggestion
  s.dependency 'nanopb', '>= 2.30908.0', '< 2.30910.0'
```

File: FirebaseMessaging.podspec
Comment (ncooke3) [inline FirebaseMessaging.podspec:68]:
```suggestion
  s.dependency 'nanopb', '>= 2.30908.0', '< 2.30910.0'
```

File: FirebasePerformance.podspec
Comment (ncooke3) [inline FirebasePerformance.podspec:69]:
```suggestion
  s.dependency 'nanopb', '>= 2.30908.0', '< 2.30910.0'
```

File: Firebase.podspec
Comment (ncooke3) [inline Firebase.podspec:2]: @paulb777, two outstanding questions:
1. Does this PR need a CHANGELOG entry for the updated min. versions for products like GTMSessionFetcher, GDT, and GULs? (My guess is no.)
2. Does the ZipBuilder need to be updated to build the latest GoogleSignIn and/or AdMob? (I assume the ZipBuilder already does so no action is needed.)

File: Gemfile
Comment (ncooke3) [inline Gemfile:19]: Revert before merging:

```suggestion
```

File: Firebase.podspec
Comment (paulb777) [reply Firebase.podspec:2]: 1. Yes for GTMSessionFetcher since that is supported for non-Google usages.
2. Should be automatic, so no.

Also, I just remembered https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseCore/Tests/Unit/FIROptionsTest.m#L629

Comment (ncooke3) [reply Firebase.podspec:2]: Got it. Also, I forget the procedure. Do I need to stage these specs for CI to pass?

File: FirebaseFirestore.podspec
Comment (ncooke3) [inline FirebaseFirestore.podspec:105]:
```suggestion
  s.dependency 'nanopb', '>= 2.30908.0', '< 2.30910.0'
```

File: FirebaseCore/Tests/Unit/FIROptionsTest.m
Comment (ncooke3) [inline FirebaseCore/Tests/Unit/FIROptionsTest.m:630]: Fix these tests before merging. I've disabled them as to unblock staging.

Comment (ncooke3) [reply FirebaseCore/Tests/Unit/FIROptionsTest.m]: Done in c3470de. I could update the comment to:
```diff
- ... when the Firebase major version hits 10.
+ ... when the Firebase major version hits 100.
```
but I have a feeling this test won't be written in Objective-C by then 😜

Comment (ncooke3) [reply FirebaseCore/Tests/Unit/FIROptionsTest.m]: Resolving because c3470de fixed the tests: https://github.com/firebase/firebase-ios-sdk/actions/runs/3106850948/jobs/5034284561

File: Firebase.podspec
Comment (ncooke3) [inline Firebase.podspec:3]: TODO: sync over deployment targets...

Comment (ncooke3) [reply Firebase.podspec:3]: TODO: CHANGELOG entry for GTMSessionFetcher

File: FirebaseFunctions.podspec
Comment (ncooke3) [inline FirebaseFunctions.podspec:1]: Looks like the `quickstart / functions` CI is failing: https://github.com/firebase/firebase-ios-sdk/actions/runs/3109715595/jobs/5040191907

It looks due to FirebaseUI neededing to be updated: https://github.com/firebase/FirebaseUI-iOS/blob/master/FirebaseAuthUI.podspec#L21

@morganchen12, is there a simple fix for this or does it require a FirebaseUI release?

**Comment by ncooke3** [comment]
@paulb777, three failures are blocking this PR. The `functions / quickstart` one is because the legacy functions quickstart uses FirebaseUI and FirebaseUI's latest version is not compatible with Firebase 10.

The other two, I'm not so sure about based on the logs.

1. [zip / package-head](https://github.com/firebase/firebase-ios-sdk/actions/runs/3109715563/jobs/5040212610)
2. [functions / quickstart](https://github.com/firebase/firebase-ios-sdk/actions/runs/3109715595/jobs/5040191907)
3. [remoteconfig / sample-build-test](https://github.com/firebase/firebase-ios-sdk/actions/runs/3109715581/jobs/5041860055)

Any ideas?

File: FirebaseFunctions.podspec
Comment (ncooke3) [reply FirebaseFunctions.podspec:1]: cc: @paulb777

File: Firebase.podspec
Comment (paulb777) [inline Firebase.podspec:105]: 11.0

File: Firebase.podspec
Comment (paulb777) [inline Firebase.podspec:107]: 12.0 here and everywhere for tvos and in CHANGELOG

**Review by paulb777 — APPROVED** [review]

File: Firebase.podspec
Comment (ncooke3) [reply Firebase.podspec]: Done in be41f6e

File: Firebase.podspec
Comment (ncooke3) [reply Firebase.podspec]: tvos: 0527f1a
changelog: 94b267c

File: FirebaseFunctions.podspec
Comment (ncooke3) [reply FirebaseFunctions.podspec:1]: Discussed offline. Will do a FirebaseUI release to relax the version requirements.

**Comment by ncooke3** [comment]
`auth / pod-lib-lint (tvos)` had a [failure](https://github.com/firebase/firebase-ios-sdk/actions/runs/3115506222/jobs/5052476923) from raising the minimum supported tvOS version. It looks due to a deprecation warning that was new in tvOS 12.

**Review by paulb777 — APPROVED** [review]
LGTM on green CI except for FirebaseUI caused failures

**Comment by ncooke3** [comment]
> `auth / pod-lib-lint (tvos)` had a [failure](https://github.com/firebase/firebase-ios-sdk/actions/runs/3115506222/jobs/5052476923) from raising the minimum supported tvOS version. It looks due to a deprecation warning that was new in tvOS 12.

Fixed in abaa752

**Comment by morganchen12** [comment]
FirebaseUI release in progress (https://github.com/firebase/FirebaseUI-iOS/pull/1100), might punt to Monday

**Comment by ncooke3** [comment]
There were some issues that arose from bumping tvOS to 12.

- [ ] On green CI, squash the tvOS commits down to one in case we want to undo bumping to tvOS 12

**Comment by ncooke3** [comment]
Ok, as of 91c7dbe, all CI is passing except for expected FirebaseUI failure.

Going to send two more commits:
- one to squash all of the tvOS 12 commits into one
- one to revert Gemfile

File: Gemfile
Comment (ncooke3) [reply Gemfile]: Done

**Comment by ncooke3** [comment]
> Ok, as of [91c7dbe](https://github.com/firebase/firebase-ios-sdk/commit/91c7dbe5c519beea872e538107261a25e6168b06), all CI is passing except for expected FirebaseUI failure.
> 
> Going to send two more commits:
> 
> * one to squash all of the tvOS 12 commits into one
> * one to revert Gemfile

I ultimately wanted one commit for all the tvOS changes in case we needed to revert in the future. Since all PR commits are squashed into `master`, I'm not sure if the squash tvOS commit could be singly reverted anyway... Anyhow, in messing with the commit history, 91c7dbe no longer exists. But the tests did pass at that point and there was no extra diff (apart from reverted Gemfile) when I submitted. So, should be good here.

**Comment by morganchen12** [comment]
FirebaseUI's been pushed, tests should pass now.

**Comment by ncooke3** [comment]
thank you @morganchen12!

**Pull request merged**
